pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -737,4 +720,36 @@ private static Mode convertRegexSubscriptionMode(RegexSubscriptionMode regexSubs
             return null;
         }
     }
+
+    private SchemaInfoProvider loadSchemaProvider(String topicName) {
+        return new MultiVersionSchemaInfoProvider(TopicName.get(topicName), this);
+    }
+
+    public LoadingCache<String, SchemaInfoProvider> getSchemaProviderLoadingCache() {
+        return schemaProviderLoadingCache;
+    }
+
+    private void schemaHandle(Schema schema, String topicName) {
+        if (schema != null && schema.supportSchemaVersioning()) {
+            SchemaInfoProvider schemaInfoProvider = null;
+            try {
+                schemaInfoProvider = this.getSchemaProviderLoadingCache().get(topicName);
+            } catch (ExecutionException e) {
```

Comment:
if it is failing to load a schema info provider, we should fail creating the subscriber or the reader.

so the code logic would be :

```
private void schemaHandle(Schema schema, String topicName) throw Throwable {
      ...
      try {
      } catch (ExceptionException e) {
            log.error("Failed to load schema info provider for topic {}", topicName, e);
            throw e.getCause();
      }
      ...
}

public <T> CompletableFuture<Reader<T>> createReaderAsync(ReaderConfigurationData<T> conf, Schema<T> schema) {
        try {
             schemaHandle(schema, conf.getTopicName();
        } catch (Throwable t) {
             return FutureUtil.failedFuture(t);
        }
        return doCreateReaderAsync(conf, schema);
}
```